nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestRenditionsObject.java

```diff
@@ -0,0 +1,222 @@
+@Deploy("org.nuxeo.ecm.platform.convert")
+@Deploy("org.nuxeo.ecm.platform.types.api")
+@Deploy("org.nuxeo.ecm.platform.types.core")
+@SuppressWarnings("unlikely-arg-type")
+public class TestRenditionsObject extends ManagementBaseTest {
+
+    @Inject
+    protected CoreSession session;
+
+    public static final String PATH = "site/management/renditions";
+
+    protected static final String ENTITY_TYPE = "entity-type";
+
+    protected static final String COMMAND_ID = "commandId";
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Inject
+    protected ThumbnailService thumbnailService;
+
+    @Test
+    public void testRendtionsPostPicturesRecompute() throws IOException {
+        // test the picture views are computed correctly and trashing them
+        DocumentRef docRef = beforePictureViewsRecompute();
+
+        // generating new picture views
+        runBulkAction("/pictures/recompute");
+
+        // checking the picture views are recomputed in the session
+        afterPicturesRecompute(docRef);
+    }
+
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.thumbnail")
+    public void testRendtionsPostThumbnailsRecomputeBulkStatus() throws IOException {
+        // check the original thumbnail is computed and trashing it
+        DocumentRef docRef = beforeThumbnailsRecompute();
+
+        // generating a new thumbnail
+        runBulkAction("/thumbnails/recompute");
+
+        // checking the thumbnail is recomputed in the session
+        afterThumbnailsRecompute(docRef);
+    }
+
+    protected JsonNode validateBulkStatus(CloseableClientResponse response) throws IOException {
+        assertEquals(HttpServletResponse.SC_OK, response.getStatus());
+        JsonNode node = mapper.readTree(response.getEntityInputStream());
+        JsonNode value = node.get(ENTITY_TYPE);
+        assertEquals(STATUS_ENTITY_TYPE, value.asText());
+        assertNotNull(node.get("state"));
+        assertNotNull(node.get("processed"));
+        assertNotNull(node.get("error"));
+        assertNotNull(node.get("errorCount"));
+        assertNotNull(node.get("total"));
+        assertNotNull(node.get("action"));
+        assertNotNull(node.get("username"));
+        assertNotNull(node.get("submitted"));
+        assertNotNull(node.get("scrollStart"));
+        assertNotNull(node.get("scrollEnd"));
+        assertNotNull(node.get("processingStart"));
+        assertNotNull(node.get("processingEnd"));
+        assertNotNull(node.get("completed"));
+        assertNotNull(node.get("processingMillis"));
+        return node;
+    }
+
+    protected void runBulkAction(String extraPath) throws IOException {
+        String fullPath = PATH + extraPath;
+        try (CloseableClientResponse response = httpClientRule.post(fullPath, null)) {
+            JsonNode result = validateBulkStatus(response);
+            JsonAssert jsonAssert = JsonAssert.on(result.toString());
+            jsonAssert.get("state").equals(State.SCHEDULED);
```

Comment:
I wonder how this can succeed, you're comparing 2 different types (String and BulkStatus#State).
`equals` probably handles it
Yet, Sonar warns with "Remove this call to "equals"; comparisons between unrelated types always return false."

Reply:
You could also simply do `assertEquals(State.SCHEDULED.name(), result.get("state").asText());` and forget about `JsonAssert` here.

Reply:
Yes I have the sonar warn, but since it works...Maybe toStringing it all down ?...
Ok for not using JsonAssert

Reply:
When the effort is low we should always take into account Sonar warnings.